subsys/shell/shell.c

```diff
@@ -28,6 +28,31 @@
 
 #define SHELL_INIT_OPTION_PRINTER	(NULL)
 
+static void shell_thread_fprintf(const struct shell *shell,
+				 enum shell_vt100_color color,
+				 const char *p_fmt, ...)
+{
+	va_list args = { 0 };
+
+	va_start(args, p_fmt);
+
+	if (IS_ENABLED(CONFIG_SHELL_VT100_COLORS) &&
```

Comment:
this if-else statement is the same for `shell_fprintf`. Could you extract it to separate function?

Reply:
done